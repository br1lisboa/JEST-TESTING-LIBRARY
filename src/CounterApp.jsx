import { useState } from 'react';
import PropTypes from 'prop-types';

export default function CounterApp({ value }) {

    const [counter, setCounter] = useState(value)

    const handleAdd = () => {
        //setCounter(counter + 1)
        setCounter((c) => c + 1)
    }
    const handleRest = () => {
        setCounter(value)
        //setCounter((c) => c = 0)
    }
    const handleSub = () => {
        //setCounter(counter + 1)
        setCounter((c) => c - 1)
    }

    return (
        <>
            <h1>CounterApp</h1>
            <h2>{counter}</h2>
            <div className='btn-container'>
                <button onClick={handleAdd}>+1</button>
                <button aria-label='btn-reset' onClick={handleRest}>Reset</button>
                <button onClick={handleSub}>-1</button>
            </div>
        </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

CounterApp.defaultProps = {
    value: 0
}