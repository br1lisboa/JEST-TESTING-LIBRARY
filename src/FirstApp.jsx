import PropTypes from 'prop-types';
//Literal react nos permite imprimir directamente variables, arrays, numbers, booleans, strings, 
//con la exepcion de objetos, los objetos no se pueden imprimir directamente. Necesitamos usar el 
//metodo JSON.stringify(nombreObjeto).

//Otra informacion interesante es crear las variables que no se necesiten rerenderizar por fuera del 
//componente.

const newMessage = {
  message: 'Hola mundo',
  title: 'Bruno'
}

function saludar() {
  console.log('funciona');
}

export const FirstApp = ({ title, subTitle, number }) => {

  //console.log(props);

  return (
    <>
      <h1 data-testid='test-title'>{title}</h1>
      <h3>{subTitle}</h3>
      <h4>{number + 100}</h4>
    </>
  );
}


FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired,
  number: PropTypes.number.isRequired
}

FirstApp.defaultProps = {
  //title: 'No hay titulo',
  subTitle: 'No hay subtitulo',
  number: 0
}