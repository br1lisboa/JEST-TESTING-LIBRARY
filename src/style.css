/* Sort line acending en F1 ordena alfabeticamente los estilos css. */

html,
body {
    background-color: #21232A;
    color: white;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}

button {
    align-items: center;
    background-color: cadetblue;
    cursor: pointer;
    display: flex;
    font-size: 1.1rem;
    height: auto;
    justify-content: center;
    margin: 10px;
    padding: 5px;
    width: 150px;
}

.btn-container {
    align-items: center;
    display: flex;
    height: auto;
    justify-content: center;
    width: 100%;
}