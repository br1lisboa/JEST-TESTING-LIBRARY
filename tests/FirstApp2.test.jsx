import { render, screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('Pruebas en FirstApp2', () => {

  const title = "Hola, soy Goku."
  const subTitle = "Soy un subtit."

  test('Debe hacer match con el snapshot', () => {

    const { container } = render(<FirstApp title={title} />)
    expect(container).toMatchSnapshot()

  })

  test('Debe de mostrar el mensaje "Hola, soy Goku."', () => {

    render(<FirstApp title={title} />)

    expect(screen.getByText(title)).toBeTruthy()
    // screen.debug() >> para ver el objeto
  })

  test('Debe de mostrar el titulo en un H1', () => {

    render(<FirstApp title={title} />)

    expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title)

  })

  test('Debe de mostrar el subtitulo enviado por props.', () => {

    render(<FirstApp title={title} subTitle={subTitle} />)

    expect(screen.getAllByText(subTitle).length).toBe(1)

  })


})