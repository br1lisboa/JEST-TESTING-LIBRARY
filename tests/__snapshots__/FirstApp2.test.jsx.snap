// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp2 Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Goku.
  </h1>
  <h3>
    No hay subtitulo
  </h3>
  <h4>
    100
  </h4>
</div>
`;
